### DISCUSSION TITLE
Is RTK opinionated about standalone actions (i.e. actions declared outside of a slice)?

### DISCUSSION THREAD

**Discussion by PeterTYLiu** [reduxjs/redux-toolkit#4703] [Q&A] [answered]
RTK gives us a straightforward way to declare actions via `createSlice`, but I can't find any docs re: best practices around standalone actions created using `createAction`.

Should we avoid standalone actions and try to always create actions via `createSlice`? If not, when are standalone actions appropriate? Does RTK even have an opinion on this matter?

**Comment by markerikson** [comment, marked as answer]
We've generally found that (fake but accurate stat) 95% of actions are only ever handled by one slice. So, `createSlice` was designed to make that use case straightforward: write a case reducer, get the action creator generated for free.

We also did work a while back to make it possible for slices to cross-import each other's actions without generally causing circular import errors.

Given that, there should be _relatively_ few times that you specifically need to use `createAction` to define an action outside of a slice.

But, it's always your codebase, and you should feel free to do that if you feel it's appropriate.
